src/server/pfs/server/driver.go

```diff
@@ -793,34 +797,40 @@ func (d *driver) getCompactedDiffFileSet(ctx context.Context, commit *pfs.Commit
 // The ProjectInfo provided to the closure is repurposed on each invocation, so it's the client's responsibility to clone the ProjectInfo if desired
 func (d *driver) listProject(ctx context.Context, cb func(*pfs.ProjectInfo) error) error {
 	authIsActive := true
-	projectInfo := &pfs.ProjectInfo{}
-	return errors.Wrap(d.projects.ReadOnly(ctx).List(projectInfo, col.DefaultOptions(), func(string) error {
-		if authIsActive {
-			resp, err := d.env.AuthServer.GetPermissions(ctx, &auth.GetPermissionsRequest{Resource: projectInfo.GetProject().AuthResource()})
-			if err != nil {
-				if errors.Is(err, auth.ErrNotActivated) {
-					// Avoid unnecessary subsequent Auth API calls.
-					authIsActive = false
-					return cb(projectInfo)
+	return errors.Wrap(dbutil.WithTx(ctx, d.env.DB, func(ctx context.Context, tx *pachsql.Tx) error {
```

Comment:
I think it would be best if listProject directly called listProjectInTransaction, rather than making the coredb calls itself.  I think it runs the risk of diverging when someone adds a special case here or there.

I suppose the wrinkle is that "InTransaction" refers to a pachyderm transaction and not a database transaction.  We probably need a new convention to allow all 3 cases; list X, list X in a sql Tx, list X in a pach Tx.  I think the right helper function could support not duplicating code and letting the caller decide; write `func (me) thatFunction(tx) { list stuff }`, and call `WithTx(tx, me.thatFunction)`/`WithPachTx(ptx, me.thatFunction)`/etc, eliminating the need for someone to type in each of these wrappers.

Reply:
For the scope of this PR I can have listProject call listProjectInTransaction, but I won't add the helpers just yet.